src/blob/blob_client_wrapper.cpp

```diff
@@ -497,7 +498,7 @@ namespace azure {  namespace storage_lite {
                     }
 
                     std::istringstream in;
-                    in.rdbuf()->pubsetbuf(buffer, length);
+                    in.str(buffer);
                     const auto blockResult = m_blobClient->upload_block_from_stream(container, blob, block_id, in, length).get();
                     free(buffer);
```

Comment:
by the time curl reads from the buffer the buffer has junk data. Is that because it's freed before being used. 
And if i use in.str(buffer) that has a different problem. We store AES encrypted data in the buffer which has null characters in the data. So cannot use in.str either. 
Any suggestions.

Reply:
Allocated memory is not necessarily be initialized with zero. memset or similar function should be used if the buffer needs initial value.